Tasks/AzureRmWebAppDeployment/deploymentProvider/PublishProfileWebAppDeploymentProvider.ts

```diff
@@ -0,0 +1,38 @@
+import { IWebAppDeploymentProvider } from "./IWebAppDeploymentProvider";
+import { TaskParameters } from "../operations/TaskParameters";
+import {PublishProfileUtility} from '../operations/PublishProfileUtility';
+import { FileTransformsUtility } from '../operations/FileTransformsUtility';
+import tl = require('vsts-task-lib/task');
+
+var packageUtility = require('webdeployment-common/packageUtility.js');
+var deployUtility = require('webdeployment-common/utility.js');
+var msDeploy = require('webdeployment-common/deployusingmsdeploy.js');
+
+export class PublishProfileWebAppDeploymentProvider implements IWebAppDeploymentProvider{
+    private taskParams: TaskParameters;
+
+    constructor(taskParams: TaskParameters) {
+        this.taskParams = taskParams;
+    }
+
+    public async PreDeploymentStep(){ }
+
+    public async DeployWebAppStep() {
+        var msDeployPublishingProfile = await PublishProfileUtility.GetTaskParametersFromPublishProfileFile(this.taskParams);
+        var webPackage = packageUtility.PackageUtility.getPackagePath(this.taskParams.Package);
+        var isFolderBasedDeployment = deployUtility.isInputPkgIsFolder(webPackage);
+        webPackage = await FileTransformsUtility.applyTransformations(webPackage, this.taskParams);
+
+        tl.debug("Performing the deployment of webapp.");
```

Comment:
nit: tl.debug("Performing the deployment of webapp using publish profile.");

Reply:
done